### DISCUSSION TITLE
Response data in...the error only

### DISCUSSION THREAD

**Discussion by kirkol** [urql-graphql/urql#2666] [Q&A] [answered]
Hello Dear Community! :)

I implemented very raw/basic Urql Client. When I call the server using useQuery hook, I see in the browser Network tab, that I receive 200 status with data I expect in the response. BUT, when I log a result of useQuery I see, that I also get the response, but it is shown as a value of "error" attribute, but "data" attribute stays undefined. Interesting is, when I use graphql-hooks package I don't face this issue. Did I do anything wrong? Any hint will be welcomed! :)

What I see in the Network tab (it's OK):
![image](https://user-images.githubusercontent.com/32139097/188168923-557ed8ce-a035-47db-bb4c-04538dabe6c4.png)

What I see in my console (not OK):
![image](https://user-images.githubusercontent.com/32139097/188169894-65422def-bbf2-4cc0-83f0-c9e0557a0c88.png)

**Comment by kitten** [comment, marked as answer]
If I'd have to guess, your API may be doing something non-compliant, e.g. not sending a proper `Content-Type` like `application/json`. 🤔
I'm not sure what you're using to get your GraphQL API to work, but if it's a custom API endpoint handler, I'd recommend using graphql helix or even graphql yoga instead

**Reply by kirkol** [reply]
It was exactly the case! ❤️ Really thank you for a great hint! Have a nice weekend! 🎉
